### PR TITLE
modules: hal_nordic: 802.15.4 carrier functions made optional

### PR DESCRIPTION
**Pull request by ahasztag**
This function adds an option to make the use of continuous carrier and modulated carrier functions optional in the Nordic 802.15.4 driver. This allows to save some memory in applications which do not use these functions

### CODE REVIEW

File: modules/hal_nordic/Kconfig [hunk 1]
```diff
@@ -205,6 +205,17 @@ endif
 
 endmenu # NRF_802154_SER_HOST || NRF_802154_SER_RADIO
 
+if (NRF_802154_RADIO_DRIVER || NRF_802154_SERIALIZATION)
+
+config NRF_802154_CARRIER_FUNCTIONS
+	bool "Enable 802.15.4 carrier functions"
+	default n
```
Comment (jciupis) [inline modules/hal_nordic/Kconfig:212]: No need to explicitly set it: https://docs.zephyrproject.org/1.14.0/guides/kconfig/index.html#redundant-defaults.

File: modules/hal_nordic/nrf_802154/CMakeLists.txt [hunk 2]
```diff
@@ -94,6 +94,8 @@ set(NRF53_SERIES  ${CONFIG_SOC_SERIES_NRF53X})
 set(SER_HOST      ${CONFIG_NRF_802154_SER_HOST})
 set(SL_OPENSOURCE ${CONFIG_NRF_802154_SL_OPENSOURCE})
 
+set(NRF_802154_CARRIER_FUNCTIONS_ENABLED ${CONFIG_NRF_802154_CARRIER_FUNCTIONS})
```
Comment (jciupis) [inline modules/hal_nordic/nrf_802154/CMakeLists.txt:97]: Shouldn't it be done using `target_compile_definitions` syntax like other configuration macros above?

**Review by jciupis — DISMISSED** [review]

**Review by jciupis — DISMISSED** [review]

File: modules/hal_nordic/Kconfig [hunk 3]
```diff
@@ -205,6 +205,16 @@ endif
 
 endmenu # NRF_802154_SER_HOST || NRF_802154_SER_RADIO
 
+if (NRF_802154_RADIO_DRIVER || NRF_802154_SERIALIZATION)
+
+config NRF_802154_CARRIER_FUNCTIONS
+	bool "Enable 802.15.4 carrier functions"
+	help
+	  This option enables functions such as modulated carrier and continuous carrier.
+	  If this option is modified, its remote counterpart must be set to the exact same value.
```
Comment (jciupis) [inline modules/hal_nordic/Kconfig:214]: `If this option is modified on a multicore SoC, ...`

Otherwise this documentation is misleading for nRF52 users.

Comment (ahasztag) [reply modules/hal_nordic/Kconfig]: Please check it now,

File: modules/hal_nordic/Kconfig [hunk 4]
```diff
@@ -205,6 +205,16 @@ endif
 
 endmenu # NRF_802154_SER_HOST || NRF_802154_SER_RADIO
 
+if (NRF_802154_RADIO_DRIVER || NRF_802154_SERIALIZATION)
```
Comment (hubertmis) [inline modules/hal_nordic/Kconfig:208]: nit: `depends on` could be cleaner than the `if` - `endif` block

**Review by hubertmis — DISMISSED** [review]

File: modules/hal_nordic/Kconfig (hunk 4 above)
Comment (ahasztag) [reply modules/hal_nordic/Kconfig]: Ok, changed

**Review by hubertmis — DISMISSED** [review]

**Review by rlubos — DISMISSED** [review]
@ahasztag Please address the compliance warning:
>  Boolean option 'NRF_802154_CARRIER_FUNCTIONS' prompt must not start with 'Enable...'. Please
check Kconfig guidelines.
 Boolean option 'NRF_802154_CARRIER_FUNCTIONS' prompt must not start with 'Enable...'. Please
check Kconfig guidelines.

Otherwise, LGTM.

File: west.yml [hunk 5]
```diff
@@ -83,7 +83,7 @@ manifest:
       groups:
         - hal
     - name: hal_nordic
-      revision: 3f7ae2320b6d348898c7b11e956a6e792d0b3724
+      revision: pull/125/head
```
Comment (rlubos) [inline west.yml:86]: Please update SHA: `6c9f23498ee6a225fee049c8c5b2e1ac760bd4a8`

And please, fix the compliance.

File: modules/hal_nordic/nrf_802154/CMakeLists.txt [hunk 6]
```diff
@@ -89,6 +89,10 @@ else()
   target_compile_definitions(zephyr-802154-interface INTERFACE NRF_802154_INTERNAL_RADIO_IRQ_HANDLING=0)
 endif()
 
+if (CONFIG_NRF_802154_CARRIER_FUNCTIONS)
+  target_compile_definitions(zephyr-802154-interface INTERFACE NRF_802154_CARRIER_FUNCTIONS_ENABLED=1)
+endif()
```
[the remark this answers is not in the document]
Comment (ahasztag) [reply modules/hal_nordic/nrf_802154/CMakeLists.txt]: Ok, went with the "else" option, as introducing the change into `nrf_802154_config.h` would be pretty complicated and would require opening a lot of PR-s. You can take a look now.

File: west.yml (hunk 5 above)
Comment (ahasztag) [reply west.yml]: Should be ok now.

**Review by maciejbaczmanski — APPROVED** [review]
LGTM. Thanks

**Review by rlubos — APPROVED** [review]

**Review by jciupis — APPROVED** [review]

**Pull request merged**